pkg/serverless/appsec/appsec.go

```diff
@@ -40,9 +40,10 @@ func New(demux aggregator.Demultiplexer) (lp *httpsec.ProxyLifecycleProcessor, e
 func NewWithShutdown(demux aggregator.Demultiplexer) (lp *httpsec.ProxyLifecycleProcessor, shutdown func(context.Context) error, err error) {
 	appsecInstance, err := newAppSec() // note that the assigned variable is in the parent scope
 	if err != nil {
-		return nil, nil, err
-	}
-	if appsecInstance == nil {
+		log.Warnf("appsec: failed to start appsec: %v -- appsec features are not available!", err)
+		// Nil-out the error so we don't incorrectly report it later on...
+		err = nil
```

Comment:
why don't we return here? what's the value of the proxy without appsec

Reply:
If the proxy does not run, the Lambda cannot work (if AppSec is configured in such a way that we even attempt starting the proxy; the runtime _IS_ proxied).